blockchain/iotex.go

```diff
@@ -0,0 +1,230 @@
+package blockchain
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/external-initiator/store"
+	"github.com/smartcontractkit/external-initiator/subscriber"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/iotexproject/iotex-proto/golang/iotexapi"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+
+	"github.com/facebookgo/clock"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	IOTX              = "iotex"
+	iotexScanInterval = 5 * time.Second
+)
+
+type iotexConnection struct {
+	sync.RWMutex
+
+	endpoint string
+	api      iotexapi.APIServiceClient
+	grpcConn *grpc.ClientConn
+}
+
+func (io *iotexConnection) connect() error {
+	io.Lock()
+	defer io.Unlock()
+	// Check if the existing connection is good.
+	if io.grpcConn != nil && io.grpcConn.GetState() != connectivity.Shutdown {
+		return nil
+	}
+	opts := []grpc.DialOption{}
+	if strings.HasSuffix(io.endpoint, "443") {
```

Comment:
This is limiting in case the endpoint is using TLS on anything other than port 443.
I see the EA has a sort of flag for secure endpoint. Can we do something similar here?

Reply:
now, i changed it to use `http/https` scheme to decide if this is a secure connection without adding additional magic. I am totally aware `grpc != http`, but i think it could a good way for operator to manage their subscriptions without additional cost.